test/pti.jl

```diff
@@ -237,3 +237,44 @@ TESTLOG = Memento.getlogger(PowerModels)
         # @test parsed_pti == pti
     end
 end
+
+
+@testset "test idempotent pti export" begin
+
+    function test_pti_idempotent(filename::AbstractString, parse_file::Function)
+        source_data = parse_file(filename)
+
+        file_tmp = "../test/data/tmp.raw"
+        PowerModels.export_pti(file_tmp, source_data)
+
+        destination_data = PowerModels.parse_file(file_tmp)
+        rm(file_tmp)
+
+        # Delete "name"  key
+        delete!(source_data, "name")
+        delete!(destination_data, "name")
+
+        @test InfrastructureModels.compare_dict(source_data, destination_data)
+    end
+
+    @testset "test parser_three_winding_test" begin
+        file = "../test/data/pti/three_winding_test.raw"
+        test_pti_idempotent(file, PowerModels.parse_file)
+    end
+
+    @testset "test case3" begin
+        file = "../test/data/pti/case3.raw"
+        test_pti_idempotent(file, PowerModels.parse_file)
+    end
+
+    # Fails at branch["5"] because I=4, J=3 -> PM parse it as f_bus=3, t_bus=4
```

Comment:
@ccoffrin, What do you think of this? The tests should skip the `source_id`?

Only if `import_all = true` we can see the value of the `MET` or `NMET` code and swap the bus numbers when writting the `.raw`.

Reply:
Good catch!

For this specific case, where PowerModels reverses the orientation of the a line, it is ok for me that the export is not ` idempotent` in terms of the original source data.  In practice this is a fairly rare occurrence.  I suggest that we leave this network out of the idempotent tests, as the other networks are available and should be sufficient for testing the functionality.

Reply:
Ok, I commented it out of the test cases.

U are right, It is rare that you need to change the orientation of the line in a `.raw` file just because the PSSE already orders them. I do not know exactly how PowerModels handle the case of metered end codes.

Reply:
At this time PowerModels ignores the metered end data so I would say that is out of the scope of this exporter.

Reply:
Ok, in the future it will be good to check what happens with different metered end data and end of line shunts data.